### PR TITLE
feat(49): Added stop_timeout parameter for locust command

### PR DESCRIPTION
**Pull request by aleks-vodolazko**
PR for https://github.com/AbdelrhmanHamouda/intensive-brew/issues/49

### CODE REVIEW

**Comment by aleks-vodolazko** [comment]
@AbdelrhmanHamouda could you take a quick look when you have time pls?

File: docs/test-config.md [hunk 1]
```diff
@@ -52,6 +52,8 @@ configurations:
       run_time: str # (1)!
       # * Test target URL 
       target_host: str
+      # * Test stop timeout
+      stop_timeout: str
```
Comment (AbdelrhmanHamouda) [inline docs/test-config.md:56]: I have few points regarding this: 

First, the option appears in the code snippet, however, it doesn't have previous reference in the section explaining what are `vanilla_specs` (i.e. line 34 through 39). I would propose adding it and mention that it is optional next to it. Example 

```
vanilla_specs: object is a mandatory ...
...
- Task timeout [Optional field]
```

Second, I know that on _locust_ side, it is called "stop timeout", however, if we think about it from this tool user perspective, "stop timeout" is a bit miss leading. My proposal is to rename it (along with its references everywhere) to something like "task_timeout"

Third, in the doc, this is mentioned as str but in the code it is referenced and used as int. I don't have specific preference but it needs to be consistent. I think `int` make more sense overall.

Forth, the comment "# * Test stop timeout" I would strongly suggest changing it to something that would communicate more about the functionality of what it does. 

The main point here is consistency and user experience.

File: docs/test-config.md [hunk 2]
```diff
@@ -192,6 +194,8 @@ configurations:
       run_time: str # (4)!
       # Test target URL 
       target_host: str # (5)!
+      # * Test stop timeout
+      stop_timeout: str
```
Comment (AbdelrhmanHamouda) [inline docs/test-config.md:198]: need to align with code and follow whatever changes based on my previous comment. Moreover, if you can kindly add an annotation # (11)! and explain in more words what it does and to what _locust_ argument it maps. This is similar to all the other options.

File: docs/test-config.md (hunk 1 above)
Comment (AbdelrhmanHamouda) [inline docs/test-config.md:56]: One additional point: there is no indication to the unit here. I know it is in seconds, but the user may not know that.

**Review by AbdelrhmanHamouda — CHANGES_REQUESTED** [review]
Thank you for your valiable contribution. As mentioned in previous communication, I think this change makes perfect sesnse and I'm more than happy to include it in the project. All and all it is good, however, it is missing some simple points. Kindly take a look and let me know.

File: docs/test-config.md (hunk 1 above)
Comment (aleks-vodolazko) [reply docs/test-config.md]: fixed

Comment (aleks-vodolazko) [reply docs/test-config.md]: Fixed

**Comment by aleks-vodolazko** [comment]
@AbdelrhmanHamouda any upd?

File: docs/test-config.md [hunk 3]
```diff
@@ -192,6 +195,8 @@ configurations:
       run_time: str # (4)!
       # Test target URL 
       target_host: str # (5)!
+      # * Timeout to end current task (sec) # (11)!
+      termination_timeout: int
```
Comment (AbdelrhmanHamouda) [inline docs/test-config.md:199]:
```suggestion
      # Timeout to end current task (sec) 
      termination_timeout: int # (11)!
```
I don't believe you desire the resultant display of that annotation 
image
As you see, it removes your comment and leave it looking bad

Comment (aleks-vodolazko) [reply docs/test-config.md:199]: oh, nice catch, thanks

**Review by AbdelrhmanHamouda — APPROVED** [review]
LGTM

**Pull request merged**